ocaml/xapi/extauth_plugin_ADpbis.ml

```diff
@@ -51,6 +55,18 @@ module Lwsmd = struct
   let start ~timeout ~wait_until_success =
     Xapi_systemctl.start ~timeout ~wait_until_success lwsmd_service
 
+  let restart ~timeout ~wait_until_success =
+    Xapi_systemctl.restart ~timeout ~wait_until_success lwsmd_service
+
+  let restart_on_error () =
+    (* Only restart once within restart_interval *)
+    let now = Unix.gettimeofday () in
+    if now -. !last_restart >= restart_interval then (
+      debug "Restart %s due to local server error" lwsmd_service ;
+      restart ~timeout:0. ~wait_until_success:false ;
+      last_restart := now
```

Comment:
I think we should set last_restart before calling `restart`, otherwise if there are N concurrent AD requests and they all fail with econnrefused we'll attempt to restart lwsmd N times, which in turn can cause more failures, and so on.
It would also be better to use monotonic clocks for measuring duration than Unix.gettimeofday, because Unix.gettimeofday can go backwards (e.g. due to NTP) which can completely disable this codepath then (if last_restart becomes a timestamp very far in the future). See usages of `Mtime` elsewhere in XAPI.